reconcile/templating/lib/rendering.py

```diff
@@ -123,11 +123,35 @@ def render_output(self) -> str:
         )
 
         if isinstance(matched_value, list):
-            if not self.template.patch.identifier:
-                raise ValueError(
-                    f"Expected identifier in patch for list at {self.template}"
-                )
-            dta_identifier = data_to_add.get(self.template.patch.identifier)
+
+            def get_identifier(data: dict[str, Any] | str) -> Any:
+                assert self.template.patch is not None  # mypy
+                if not self.template.patch.identifier:
+                    raise ValueError(
+                        f"Expected identifier in patch for list at {self.template}"
+                    )
+                if isinstance(data, str):
+                    # if the data is already a string, use it as is.
+                    # this is the case if the target is a list of strings
+                    if self.template.patch.identifier != ".":
+                        raise ValueError(
+                            f"Use patch.identifier '.' for updating list of strings at {self.template}"
+                        )
+                    return data
+
+                if self.template.patch.identifier.startswith("$."):
+                    # jsonpath support
+                    if matches := [
+                        match.value
+                        for match in parse_jsonpath(
+                            self.template.patch.identifier
+                        ).find(data)
+                    ]:
+                        return matches[0]
+                    return None
+                return data.get(self.template.patch.identifier)
```

Comment:
can we align both to use json path? like if identifier is `$`, it can handle both list of string and normal json path.

Reply:
I don't get it. For lists, do you want `identifier: $.` instead of `identifier: .`?

Reply:
just `identifier: $`, then let jsonpath support branch handle it.

Reply:
But `identifier` isn't used for lists.

Reply:
it can be used, update test fixture https://github.com/app-sre/qontract-reconcile/pull/4487/files#diff-9c849344aa377c6ae1b6b04967c3d28e203c7334bb57bd65c526d7ff3ce0576cR10 to `identifier: $`, then remove `str` check https://github.com/app-sre/qontract-reconcile/pull/4487/files#diff-29d1a22f3899b8251e82525a326a71f6be4c31d519ef497d3ca233f557a55630R133-R140, update json path condition to `.startswith("$")`, test pass.

Reply:
Awesome! 🙇 Applied in the last commit.